homeassistant/components/caldav/todo.py

```diff
@@ -92,3 +111,71 @@ async def async_update(self) -> None:
             for resource in results
             if (todo_item := _todo_item(resource)) is not None
         ]
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+        try:
+            await self.hass.async_add_executor_job(
+                partial(
+                    self._calendar.save_todo,
+                    summary=item.summary,
+                    status=TODO_STATUS_MAP_INV.get(
+                        item.status or TodoItemStatus.NEEDS_ACTION, "NEEDS-ACTION"
+                    ),
+                ),
+            )
+        except (requests.ConnectionError, DAVError) as err:
+            raise HomeAssistantError(f"CalDAV save error: {err}") from err
+
+        self.async_schedule_update_ha_state(force_refresh=True)
+
+    async def async_update_todo_item(self, item: TodoItem) -> None:
+        """Update a To-do item."""
+        uid: str = cast(str, item.uid)
+        try:
+            todo = await self.hass.async_add_executor_job(
+                self._calendar.todo_by_uid, uid
+            )
+        except NotFoundError as err:
+            raise HomeAssistantError(f"Could not find To-do item {uid}: {err}") from err
```

Comment:
any reason to include the `err` in the message? Or are there different ways of finding an item.